### PR TITLE
Radix config

### PR DESCRIPTION
**Pull request by davidramsay95**

### CODE REVIEW

File: Equinor.Maintenance.API.EventEnhancer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RUN  dotnet restore . /property:Configuration=Release
+
+COPY . .
+RUN dotnet publish . -c Release --output ./out
+
+FROM mcr.microsoft.com/dotnet/aspnet:6.0-bullseye-slim-amd64
+WORKDIR /app
+COPY --from=build-env /publish/out .
+
+ENV USE_ENVIRONMENT_CREDENTIAL true
```
Comment (hekka) [inline Equinor.Maintenance.API.EventEnhancer/Dockerfile:15]:
```suggestion
```

File: Equinor.Maintenance.API.EventEnhancer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ENV USE_RADIX_CERT_HEADERNAME true
```
Comment (hekka) [inline Equinor.Maintenance.API.EventEnhancer/Dockerfile:16]:
```suggestion
```

File: Equinor.Maintenance.API.EventEnhancer/appsettings.json [hunk 3]
```diff
@@ -8,16 +8,19 @@
   "AzureAd": {
     "TenantId": "3aa4a235-b6e2-48d5-9195-7fcf05b459b0",
     "Instance": "https://login.microsoftonline.com/",
-    "ClientId": "",
+    "ClientId": "864ad7ca-19d4-46a9-94e9-5e5baa231e35",
```
Comment (hekka) [inline Equinor.Maintenance.API.EventEnhancer/appsettings.json:11]:
```suggestion
    "ClientId": "",

```

File: radixconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      publicPort: http
+      ports:
+        - name: http
+          port: 5000
+      secrets:
+        - ClientSecret
+      ingressConfiguration:
+        - leastconnectedlb
+      environmentConfig:
+        - environment: test
```
Comment (davidramsay95) [inline radixconfig.yaml:26]:
```suggestion
        - environment: development
```

File: Equinor.Maintenance.API.EventEnhancer/appsettings.json [hunk 5]
```diff
@@ -8,16 +8,19 @@
   "AzureAd": {
     "TenantId": "3aa4a235-b6e2-48d5-9195-7fcf05b459b0",
     "Instance": "https://login.microsoftonline.com/",
-    "ClientId": "",
+    "ClientId": "864ad7ca-19d4-46a9-94e9-5e5baa231e35",
     "ClientSecret": "",
     "ClientCertificates": [],
-    "AllowedWebHookOrigins": []
+    "AllowedWebHookOrigins": [
+      "My Allowed Origin",
```
Comment (hekka) [inline Equinor.Maintenance.API.EventEnhancer/appsettings.json:15]: We should load these from key vault

File: Equinor.Maintenance.API.EventEnhancer/appsettings.json [hunk 6]
```diff
@@ -8,16 +8,19 @@
...
+      "Another Allowed Origin"
+    ]
   },
   "ConnectionStrings": {
     "ServiceBus": "",
     "ApplicationInsights" : "",
-    "KeyVault": "",
-    "MaintenanceApi":""
+    "MaintenanceApi": "https://api-test.gateway.equinor.com/maintenance-api",
```
Comment (hekka) [inline Equinor.Maintenance.API.EventEnhancer/appsettings.json:22]:
```suggestion
    "MaintenanceApi": "",

```

File: Equinor.Maintenance.API.EventEnhancer/appsettings.json [hunk 7]
```diff
@@ -8,16 +8,19 @@
...
+    "KeyVault": "https://s390-mapi-event-kv-sbx.vault.azure.net/"
```
Comment (hekka) [inline Equinor.Maintenance.API.EventEnhancer/appsettings.json:23]:
```suggestion
    "KeyVault": ""

```

File: radixconfig.yaml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      build:
+        from: hf/**/*
+    - name: prod
+  components:
+    - name: api
+      dockerfileName: Equinor.Maintenance.API.EventEnhancer/Dockerfile
+      publicPort: http
+      ports:
+        - name: http
+          port: 5000
```
Comment (davidramsay95) [inline radixconfig.yaml:20]:
```suggestion
          port: 5005
```

File: radixconfig.yaml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      ingressConfiguration:
+        - leastconnectedlb
+      environmentConfig:
+        - environment: test
+          authentication:
+            clientCertificate:
+              verification: "on"
+              passCertificateToUpstream: true
+          variables:
+            ASPNETCORE_ENVIRONMENT: "test"
```
Comment (hekka) [inline radixconfig.yaml:32]:
```suggestion
            ASPNETCORE_ENVIRONMENT: "Development"
```

File: radixconfig.yaml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        - name: http
+          port: 5000
+      secrets:
+        - ClientSecret
+      ingressConfiguration:
+        - leastconnectedlb
+      environmentConfig:
+        - environment: test
+          authentication:
+            clientCertificate:
```
Comment (hekka) [inline radixconfig.yaml:28]: bo clientCertificate validation (yet..)

File: radixconfig.yaml (hunk 4 above)
Comment (hekka) [inline radixconfig.yaml]:
```suggestion
        - environment: development
```

File: radixconfig.yaml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  name: maintenance-api-event-enhancer
+spec:
+  environments:
+    - name: development
+      build:
+        from: main
+    - name: qa
+      build:
+        from: hf/**/*
+    - name: prod
```
Comment (hekka) [inline radixconfig.yaml:13]:
```suggestion
```

File: radixconfig.yaml [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+  components:
+    - name: api
+      dockerfileName: Equinor.Maintenance.API.EventEnhancer/Dockerfile
+      publicPort: http
+      ports:
+        - name: http
+          port: 5000
+      secrets:
+        - ClientSecret
```
Comment (hekka) [inline radixconfig.yaml:22]:
```suggestion
        - AzureAd__ClientSecret
```

File: radixconfig.yaml [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+            requests:
+              cpu: "500m"
+            limits:
+              memory: "2Gi"
+          authentication:
+            clientCertificate:
+              verification: "on"
+              passCertificateToUpstream: true
+          variables:
+            ASPNETCORE_ENVIRONMENT: "prod"
```
Comment (hekka) [inline radixconfig.yaml:62]:
```suggestion
    
```

**Review by hekka — COMMENTED** [review]
something like this I think

**Pull request merged**